node.js

```diff
@@ -48,7 +48,44 @@ module.exports = {
     'space-before-function-paren': off,
     'keyword-spacing': error,
     'eol-last': [ error, always ],
-    'brace-style': [ error, '1tbs', { allowSingleLine: true } ],
-    'curly': [ error, all ]
+    'brace-style': [ error, '1tbs', { allowSingleLine: false } ],
+    'curly': [ error, all ],
+    'block-spacing': [ error,  always ],
+    'indent': [ error, 2, { SwitchCase: 1 } ],
+    'no-tabs': error,
+    'spaced-comment': [ error, always ],
+    'array-callback-return': warn,
+    'eqeqeq': [ error, always ],
```

Comment:
error, smart

Reply:
why do you want smart? it's much better to simply be consistent on it

Reply:
Because I'm using flow and `a == null` and `a === null` has different meaning for typechecking.

Reply:
I consider `== null` to be an antipattern - types should be stricter and it's magic that isn't clearly known when another developer reads it

If it's undefined, use `typeof a === 'undefined'`, if it's null, use `a === null`, if it's either (which it usually shouldn't be), put them together.

It's similar to using `~a.indexOf(b)` instead of `a.indexOf(b) !== -1`, avoiding magic is generally  better (ignoring `.includes` for this example)

Again, can remove and separate into different PR/discussion if it's contentious

Reply:
`typeof foo !== 'undefined' && foo !== null` is too verbose compared to `foo != null`, especially when you have to check multiple attributes in object for example; why not introducing it as `error, smart` and opening new pr with stricter rule we can vote on? I don't think anybody is going argue against `error, smart` and it handles all cases except one with null.

Reply:
sorry, correction, not smart but:
```
[ "error", "always", { null: "ignore" } ]
```